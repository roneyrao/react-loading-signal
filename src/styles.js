export { default } from './styles/LoadingSignal.css';
