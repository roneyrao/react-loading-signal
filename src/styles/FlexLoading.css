.loading {
	width:100%;
	top:50%;
	transform: translateY(-50%);
	z-index:999;
	visibility:hidden;
	text-align:center;
	word-wrap:normal;
	white-space:nowrap;
}
.masked{
	top:0;
	right:0;
	bottom:0;
	left:0;
	z-index:9999;
	display:flex;
	justify-content:center;
	align-items:center;
	transform: none;
}
.masked:before{
	content:'';
	top:0;
	right:0;
	bottom:0;
	left:0;
	background:rgba(150, 150, 150, 0.4);
	position: absolute;
}
.masked.loading>* {
	transform: none;
	z-index: 1;
}

.local{
	composes:loading;
	position:absolute;
}
.localMasked{
	composes:local masked;
}

.global{
	composes:loading;
	position:fixed;
}
.globalMasked{
	composes:global masked;
}

.messageList{
	margin-top:1em;
	line-height:2em;
}
/*theme 'blobs'*/
.blobs{
    width: 100%;
	height:1em;
	background:url(./images/blobs.gif) no-repeat center center;
	background-size:auto 100%;
}
/*theme 'spinning'*/
.spinning{
	min-width:3em;
    padding: 4.5em 1.5em 1.5em;
    border-radius: 0.8em;
	background:rgba(0,0,0,.7) url(./images/spinning.gif) no-repeat center 1.5em;
	background-size:auto 3em;
	color: white;
}





