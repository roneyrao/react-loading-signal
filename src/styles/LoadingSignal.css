.loading {
  top: 50%;
  left: 50%;
  z-index: 999;
  visibility: hidden;
  text-align:center;
  transform: translate(-50%, -50%);
}
.masked:before {
  content: '';
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(150, 150, 150, 0.4);
  position: absolute;
}
.masked.loading > * {
  transform: none;
  z-index: 1;
}

.local {
  composes: loading;
  position: absolute;
}

.global {
  composes: loading;
  position: fixed;
  min-width: 3em;
  max-width: 50%;
}

.masked {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 9999;
  display: flex;
  justify-content: center;
  align-items: center;
  transform: none;
  min-width: auto;
  max-width: none;
}
.globalMasked {
  composes: global masked;
}
.localMasked {
  composes: local masked;
}

.messageList {
  line-height: 1em;
  list-style: none;
  margin:0;
  padding:0;
}
.messageList > li {
  padding-top: 1em;
}
/*theme 'blobs'*/
.blobs {
  width: 12em;
  height: 1em;
  background: url(./images/blobs.gif) no-repeat center center;
  background-size: auto 100%;
}
/*theme 'spinning'*/
.spinning {
  padding: 4.5em 1.5em 1.5em;
  border-radius: 0.8em;
  background: rgba(0, 0, 0, 0.7) url(./images/spinning.gif) no-repeat center 1.5em;
  background-size: auto 3em;
  color: white;
  text-align: center;
}
/*theme 'circling'*/
.circling {
}
.circling:before {
  content:'';
  display: block;
  margin:0 auto;
  width: 2em;
  height: 2em;
  border-radius: 1em;
  box-shadow: 0 0.1em 0 0 #000000;
  animation: circling 1s linear infinite;
}
@keyframes circling {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
/*theme 'progress'*/
.progress {
  background: rgba(255, 255, 255, 0.8);
  padding: 10px;
  border-radius: 5px;
  box-shadow: 0px 0px 5px;
}
.progressBar {
  height: 5px;
  border: 1px solid #057927;
  margin-top:1em;
}
.progressBar > div{
  width: 0%;
  height: 100%;
  background: rgb(5, 121, 39);
}
