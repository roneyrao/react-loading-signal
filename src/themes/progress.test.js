import React from 'react';
import { shallow } from 'enzyme';
import Progress from './progress';
import styles from '../styles';

test('no caption', () => {
  const msg = {
    progress: 0.5,
  };
  const wrapper = shallow(<Progress message={msg} />);
  expect(wrapper.find(`.${styles.progress}`).text()).toBe('progress: 50.00%');
  expect(wrapper.find(`.${styles.progressBar}`).childAt(0).prop('style').width).toBe('50%');
  expect(wrapper).toMatchSnapshot();
});
test('custom caption', () => {
  const captionStr = 'adfasfaf';
  const msg = {
    progress: 0.5,
    caption: jest.fn(() => captionStr),
  };
  const wrapper = shallow(<Progress message={msg} />);
  expect(wrapper.find(`.${styles.progress}`).text()).toBe(captionStr);
  expect(msg.caption).toBeCalledWith(msg.progress);
  expect(wrapper.find(`.${styles.progressBar}`).childAt(0).prop('style').width).toBe('50%');
});
