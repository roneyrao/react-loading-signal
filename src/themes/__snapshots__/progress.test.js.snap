// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no caption 1`] = `
<div
  className="LoadingSignal__progress"
>
  <div>
    progress: 50.00%
  </div>
  <div
    className="LoadingSignal__progressBar"
  >
    <div
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
</div>
`;
